In [1]:
import pandas as pd

In [3]:
features_df = pd.read_csv('../dataset/Features data set.csv')
sales_df = pd.read_csv('../dataset/sales data-set.csv')
stores_df = pd.read_csv('../dataset/stores data-set.csv')
print("All Dataset Loaded Successfully!!!!")

All Dataset Loaded Successfully!!!!


In [4]:
# Shape and Preview of features_df dataset
print("Features Data Set Shape:- ", features_df.shape)
print("\n",features_df.head())

# Shape and Preview of sales_df dataset
print("\nSales Data Set Shape:- ", sales_df.shape)
print("\n",sales_df.head())

# Shape and Preview of features_df dataset
print("\nStores Data Set Shape:- ", stores_df.shape)
print("\n",stores_df.head())

Features Data Set Shape:-  (8190, 12)

    Store        Date  Temperature  Fuel_Price  MarkDown1  MarkDown2  \
0      1  05/02/2010        42.31       2.572        NaN        NaN   
1      1  12/02/2010        38.51       2.548        NaN        NaN   
2      1  19/02/2010        39.93       2.514        NaN        NaN   
3      1  26/02/2010        46.63       2.561        NaN        NaN   
4      1  05/03/2010        46.50       2.625        NaN        NaN   

   MarkDown3  MarkDown4  MarkDown5         CPI  Unemployment  IsHoliday  
0        NaN        NaN        NaN  211.096358         8.106      False  
1        NaN        NaN        NaN  211.242170         8.106       True  
2        NaN        NaN        NaN  211.289143         8.106      False  
3        NaN        NaN        NaN  211.319643         8.106      False  
4        NaN        NaN        NaN  211.350143         8.106      False  

Sales Data Set Shape:-  (421570, 5)

    Store  Dept        Date  Weekly_Sales  IsHolida

In [5]:
# check missing values in the dataset.
print("\nMissing Values:")
print("\nFeatures Table")
print(features_df.isna().sum())
print("\nSales :- ")
print(sales_df.isna().sum())
print("\nStores :- ")
print(stores_df.isna().sum())


Missing Values:

Features Table
Store              0
Date               0
Temperature        0
Fuel_Price         0
MarkDown1       4158
MarkDown2       5269
MarkDown3       4577
MarkDown4       4726
MarkDown5       4140
CPI              585
Unemployment     585
IsHoliday          0
dtype: int64

Sales :- 
Store           0
Dept            0
Date            0
Weekly_Sales    0
IsHoliday       0
dtype: int64

Stores :- 
Store    0
Type     0
Size     0
dtype: int64


In [6]:
# Data Summary Points
print("--"*75)
print("\nFeatures Table")
print(features_df.info())
print("--"*75)
print("\nSales Table")
print(sales_df.info())
print("--"*75)
print("\nStores Table")
print(stores_df.info())
print("--"*75)

------------------------------------------------------------------------------------------------------------------------------------------------------

Features Table
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8190 entries, 0 to 8189
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Store         8190 non-null   int64  
 1   Date          8190 non-null   object 
 2   Temperature   8190 non-null   float64
 3   Fuel_Price    8190 non-null   float64
 4   MarkDown1     4032 non-null   float64
 5   MarkDown2     2921 non-null   float64
 6   MarkDown3     3613 non-null   float64
 7   MarkDown4     3464 non-null   float64
 8   MarkDown5     4050 non-null   float64
 9   CPI           7605 non-null   float64
 10  Unemployment  7605 non-null   float64
 11  IsHoliday     8190 non-null   bool   
dtypes: bool(1), float64(9), int64(1), object(1)
memory usage: 712.0+ KB
None
----------------------------------------------

In [8]:
# Statistical Summary Points
print("--"*75)
print("\nFeatures Table")
print(features_df.describe(include='all'))
print("--"*75)
print("\nSales Table")
print(sales_df.describe(include='all'))
print("--"*75)
print("\nStores Table")
print(stores_df.describe(include='all'))
print("--"*75)

------------------------------------------------------------------------------------------------------------------------------------------------------

Features Table
              Store        Date  Temperature   Fuel_Price      MarkDown1  \
count   8190.000000        8190  8190.000000  8190.000000    4032.000000   
unique          NaN         182          NaN          NaN            NaN   
top             NaN  05/02/2010          NaN          NaN            NaN   
freq            NaN          45          NaN          NaN            NaN   
mean      23.000000         NaN    59.356198     3.405992    7032.371786   
std       12.987966         NaN    18.678607     0.431337    9262.747448   
min        1.000000         NaN    -7.290000     2.472000   -2781.450000   
25%       12.000000         NaN    45.902500     3.041000    1577.532500   
50%       23.000000         NaN    60.710000     3.513000    4743.580000   
75%       34.000000         NaN    73.880000     3.743000    8923.310000 

## Insights from Data Inspection

#### 1) Features DataSet(8190 rows, 12 columns)
   - Markdown 1-5 -> ~50% missing values(expected that campaigns don't run every week)
   - CPI & Unemployment -> 585 missing values(~7%)
   - 'Date' column need to be converted from object to date format.
     
   - **Temperature**
        - Avg ~59°F, range –7°F (very cold) to 102°F (very hot).
        - Stores operate in very diverse climates → sales may depend on weather (e.g., hot weather = higher beverage/AC sales).

   - **Fuel Price**
        - Range: $2.47 – $4.47.
        - Fuel cost fluctuations may impact customer spending power → sales might dip when fuel is expensive.

   - **MarkDown1–5 (promotional discounts)**
        - Extremely skewed with huge outliers (Max up to 771,448) → only active in some weeks, absent in many (missing values).
        - Promotions likely drive sales spikes → important predictor.

   - **CPI (Consumer Price Index)**
        - Range: 126 – 229, std ~39 → stores span different economic environments.
        - Higher CPI → higher living costs → may reduce discretionary purchases.

   - **Unemployment**
        - Range: 3.68% – 14.31%, mean ~7.8%.
        - Big spread → local economy strongly influences store sales.

   - **IsHoliday**
        - 7605 False vs 585 True → ~7% are holidays.
        - Sales spikes should align with these holiday weeks.


#### 2) **Sales Dataset(421570 rows, 5 columns)**

   - No missing values.
   - Columns - Store, Dept, Date, Weekly_Sales, IsHoliday.
   - Sales are by Store + Dept + Date
   - Date column aligns with features dataset so will be key for merging.
   - **Weekly Sales**
     - Mean = $15.9k, but max = $693k, min = –$4.9k (returns/refunds).
     - Very high variance → model must handle skewness.
     - Median ~$7.6k → sales distribution is right-skewed (few very high values).

   - **Seasonality**
     - Top date is 23-Dec-2011, right before Christmas → confirms strong holiday seasonality.
     - Holiday effects are crucial features.

   - **Store / Dept**
     - Stores 1–45, Depts 1–99 → not all departments are equally profitable.
     - Some departments likely dominate (e.g., groceries).
   
#### 4) **Stores Dataset(45 rows, 3 columns)**

   - No missing values.
   - Columns: Store, Type(A/B/C), Size.
   - **Store Size**
     - Avg ~130k sq.ft., range 34k – 219k.
     - Larger stores (likely “Type A”) may generate more sales.
   - **Type**
     - Three categories: A, B, C.
     - A most common (22 stores).
     - Likely: Type A = supercenters, Type B = medium, Type C = smaller.